e2e/test/scenarios/admin-2/whitelabel.cy.spec.js

```diff
@@ -151,6 +157,113 @@ describeEE("formatting > whitelabel", () => {
       cy.get("body").should("have.css", "font-family", `"${font}", sans-serif`);
     });
   });
+
+  describe("Help link", () => {
+    beforeEach(() => {
+      cy.intercept("PUT", "/api/setting/help-link").as("putHelpLink");
+      cy.intercept("PUT", "/api/setting/help-link-custom-destination").as(
+        "putHelpLinkUrl",
+      );
+    });
+
+    it("should allow customising the help link", () => {
+      cy.log("Hide Help link");
+
+      cy.signInAsAdmin();
+      cy.visit("/admin/settings/whitelabel");
+
+      cy.findByLabelText("Link to Metabase help").should("be.checked");
+
+      cy.findByTestId("help-link-setting").findByText("Hide it").click();
+      cy.wait("@putHelpLink");
+
+      cy.signInAsNormalUser();
+
+      cy.visit("/");
+      openSettingsMenu();
+      helpLink().should("not.exist");
+
+      cy.log("Set custom Help link");
+
+      cy.signInAsAdmin();
+      cy.visit("/admin/settings/whitelabel");
+
+      cy.findByTestId("help-link-setting")
+        .findByText("Go to a custom destination...")
+        .click();
+
+      cy.findByLabelText("Help link custom destination")
+        .should("have.focus")
+        .type("https://example.org/custom-destination")
+        .blur();
+
+      cy.wait("@putHelpLinkUrl");
+
+      cy.wait("@putHelpLink");
+
+      cy.log("Check that on page load the text field is not focused");
+      cy.reload();
+
+      cy.findByLabelText("Help link custom destination").should(
+        "not.have.focus",
+      );
+
+      cy.signInAsNormalUser();
+      cy.visit("/");
+      openSettingsMenu();
+      helpLink().should(
+        "have.attr",
+        "href",
+        "https://example.org/custom-destination",
+      );
+
+      cy.log("Set default Help link");
+
+      cy.signInAsAdmin();
+      cy.visit("/admin/settings/whitelabel");
+
+      cy.findByTestId("help-link-setting")
+        .findByText("Link to Metabase help")
+        .click();
+
+      cy.wait("@putHelpLink");
+
+      cy.visit("/");
+      openSettingsMenu();
+
+      helpLink()
+        .should("have.attr", "href")
+        .and("include", "https://www.metabase.com/help-premium?");
+
+      cy.signInAsNormalUser();
+      cy.visit("/");
+      openSettingsMenu();
+
+      helpLink()
+        .should("have.attr", "href")
+        .and("include", "https://www.metabase.com/help?");
+    });
+
+    it("should not create a race condition when the custom url input is selected and we click on the radio", () => {
```

Comment:
This wasn't quite the test for the race condition because it's missing the last step which is to select the custom URL again without changing the URL.

1. default URL by default
2. Select custom URL
3. Type URL
4. Select hide it
5. Select custom URL again <- this step was missing.

Reply:
Thanks, interestingly enough, I reproduced an issue even without the last step and assumed it was caused by the same thing.
I added a test case also for your case (just added the step and remove the cy.wait), let's keep them both until we are 100% about having 1 or 2 settings, as going with the merged setting could have both those issues.